user-contrib/Ltac2/tac2entries.ml

```diff
@@ -911,25 +911,19 @@ let print_ltac qid =
 
 (** Calling tactics *)
 
-let solve ~pstate default tac =
-  let pstate, status = Declare.Proof.map_fold_endline pstate ~f:(fun etac p ->
-    let with_end_tac = if default then Some etac else None in
-    let g = Goal_select.get_default_goal_selector () in
-    let (p, status) = Proof.solve g None tac ?with_end_tac p in
-    (* in case a strict subtree was completed,
-       go back to the top of the prooftree *)
-    let p = Proof.maximal_unfocus Vernacentries.command_focus p in
-    p, status)
-  in
-  if not status then Feedback.feedback Feedback.AddedAxiom;
-  pstate
-
-let call ~pstate ~default e =
+let ltac2_interp e =
   let loc = e.loc in
   let (e, t) = intern ~strict:false [] e in
   let () = check_unit ?loc t in
```

Comment:
Perhaps the intern + check_unit work should be done outside the comtactic intepreter. If we add `par:` support this would avoid doing it once per goal. AFAICT the intern result is marshallable so that wouldn't be a problem.